### ISSUE TITLE
Migrate to notion public API

### ISSUE THREAD

**Issue by pi0** [janniks/vue-notion#32]
Hi @janniks. Thanks for all the awesome efforts you put into this package and nuxt support.

If you've followed notion news ([tweet](https://twitter.com/NotionHQ/status/1392883313757409284)) there is now a public API and [notion-sdk-js](https://github.com/makenotion/notion-sdk-js) (pending for universal support https://github.com/makenotion/notion-sdk-js/issues/38). I'm wondering how is your timing and take on this. And if needed any help would be more than happy to work on a PR.

**Comment by janniks** [comment]
Yes, definitely up for it (happy to take PRs as well). It was always the plan to work with the official APIs at some point. 

Although I will also check what react-notion is doing regarding this — it's nice to copy from "upstream".

**Comment by janniks** [comment]
The current beta API from differs a lot form the splitbee notion-api-worker. This is likely a longer issue.

If anybody wants to tackle this issue — hit me up!

**Comment by xianbinKit** [comment]
@janniks hello

 I really want to use Notion official API with vue-notion, I have less experience, but I want to try.  where you are in this subject ?  can we talk ?

**Comment by janniks** [comment]
Hi, sure! I don't really know anything about the new API. I looked at the rough structure, and it looked very different from the splitbee/notion-api-worker — that's when I gave up 😅

It would start with [retrieving](https://developers.notion.com/reference/get-block-children) the (child) blocks of a page (which is also a ~block). Then use this as a replacement for `vue-notion` `blockMap`. And then changes have to be made everywhere, where the unofficial `blockMap` differs from the block children response of the official API.

**Comment by xianbinKit** [comment]
Hi @janniks 

I make a test on top of Vue Notion.



the test code is here : https://github.com/xianbinKit/vue-notion

to do a test : 

`export NOTION_SECRET=YOUR_NOTIOIN_SECRET_KEY;npm run notionServer`

the demo json result is here :  https://github.com/xianbinKit/vue-notion/blob/main/notionServer/demoPage.json

and the Notion Page is here : https://wishcompta.notion.site/Public-Test-Notion-Page-36baad2799494b13aa09f7d2a66b2561




the demo json result is more Notion-type-like structure,  do you think you can change Vue Notion for this structure json ? 

Thank you

**Comment by janniks** [comment]
Hi 👋🏻 Thanks for the testing.
Unfortunately, the public API does not yet support as much as the private API. It will take a while to implement this — sorry

**Comment by aromeronavia** [comment]
What I see is that the Public API is more focused on integrations around workflows (when Notion database item is created, do some work in This other platform). I do think using the private API is more useful for this use case

**Comment by ebanDev** [comment]
Hi there ! It's been more than a year, anything new about the public API usage ? :) I think that would be a really interesting feature

**Comment by ali4zimi** [comment]
Hi @pi0 @janniks 
I’d be happy to help with adding support for the official Notion API. I’ve already done some initial experiments and confirmed that the official API can be used to render basic blocks successfully.

My proposal would be to keep the current legacy implementation intact for backward compatibility, and develop the official API support as a separate, optional implementation (for example via a new module or branch).

Before going further, I just wanted to check if this direction makes sense for the project and whether you’d be open to contributions like this.

**Comment by janniks** [comment]
Wonderful to hear! Yes, I'm open to it. I think it would be great to have all legacy code still as is and add a nee export NotionRendererOfficial (or better name) and have the official API use that. We can add this as a feature release or breaking release. Either works fine, but I'd want old code to keep working.

Happy to coordinate more and work on architectural stuff as well 🚀

**Comment by ali4zimi** [comment]
Thanks @janniks 
I’ve already started drafting something, but I wanted to ask about one design decision first.
For official API support, would you rather:
- include **@notionhq/client** and have users pass an **API key + page ID**, or
- keep the renderer focused on rendering and let users fetch the data themselves and pass it in (with docs explaining how)?
